locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/JcloudsLocationSecurityGroupEditor.java

```diff
@@ -0,0 +1,303 @@
+    private final Predicate<Exception> isExceptionRetryable;
+
+    /**
+     * Constructor for editor that will retry operations upon exceptions.
+     * @param location JClouds location where security groups will be managed.
+     * @param computeService The JClouds compute service instance.
+     * @param predicate A predicate indicating whether the customiser can retry a request (e.g. to add a security group
+     * or a rule) if the attempted operation throws a Throwable.
+     */
+    public JcloudsLocationSecurityGroupEditor(Location location, ComputeService computeService, Predicate predicate) {
+        this.location = location;
+        this.computeService = computeService;
+        this.locationId = this.computeService.getContext().unwrap().getId();
+        this.securityApi = this.computeService.getSecurityGroupExtension(); // TODO surely check for isPresent else throw?
+        this.isExceptionRetryable = predicate;
+    }
+
+    /**
+     * Constructor for editor that never retries requests if the attempted operation fails.
+     * @param location JClouds location where security groups will be managed.
+     * @param computeService The JClouds compute service instance.
+     */
+    public JcloudsLocationSecurityGroupEditor(Location location, ComputeService computeService) {
+        this(location, computeService, Predicates.alwaysFalse());
+    }
+
+
+    /**
+     * Flag to indicate whether the given compute service has support for security groups.
+     */
+    public boolean hasServiceSupport() {
+        return securityApi.isPresent();
+    }
+
+    /**
+     * Get the location in which security groups will be created or searched.
+     */
+    public Location getLocation() {
+        return location;
+    }
+
+    /**
+     * Get the location id from the compute service (e.g. "aws-ec2").
+     */
+    public String getLocationId() {
+        return locationId;
+    }
+
+    public Set<SecurityGroup> getSecurityGroupsForNode(final String nodeId) {
+        return securityApi.get().listSecurityGroupsForNode(nodeId);
+    }
+
+    /**
+     * Create the security group. As we use jclouds, groups are created with names prefixed
+     * with {@link #JCLOUDS_PREFIX}. This method is idempotent.
+     * @param name Name of the group to create
+     * @return The created group.
+     */
+    public SecurityGroup createSecurityGroup(final String name) {
+
+        LOG.debug("Creating security group {} in {}", name, location);
+        Callable<SecurityGroup> callable = new Callable<SecurityGroup>() {
+            @Override
+            public SecurityGroup call() throws Exception {
+                return securityApi.get().createSecurityGroup(name, location);
+            }
+        };
+        return runOperationWithRetry(callable);
+    }
+
+    /**
+     * Removes a security group and its permissions.
+     * @return true if the group was found and removed.
+     */
+    public Boolean removeSecurityGroup(final SecurityGroup group) {
```

Comment:
Personal preference for returning `boolean` if possible - under what circumstances would it return null?